uploader/push_to_sling.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/python3
+import os
+import requests
+import getpass
+import argparse
+import getpass
+from pathlib import Path
+from datetime import datetime
+
+HEADERS = {'cache-control': 'no-cache',
+           'Accept': 'application/json'}
+
+parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter, description='''\
+Bulk upload module for Pantheon 2. This tool will scan a directory recursively and upload relevant files. Relevancy is determined by examining the filename.
+
+If the file suffix is .adoc or .asciidoc, then it is relevant if the extended suffix is one of these:
+  .title.adoc
+  .module.adoc
+  .internal.adoc
+For example: hotNewProduct.title.adoc
+
+If the file suffix is anything else, then the file is relevant unless the extended suffix contains .ignore, like so:
+  placeholderArt.ignore.jpg'
+ 
+''')
+parser.add_argument('--server', '-s', help='The Pantheon server to upload modules to, default http://localhost:8080', default='http://localhost:8080')
+parser.add_argument('--repository', '-r', help='The name of the repository; this is a convenience for the user. If not specified, one will be autogenerated', default=getpass.getuser() + '_' + datetime.now().strftime('%Y-%m-%d_%H-%M-%S'))
+parser.add_argument('--user', '-u', help='Username for authentication, default \'demo\'', default='demo')
+parser.add_argument('--password', '-p', help='Password for authentication, default \'demo\'. If \'-\' is supplied, the script will prompt for the password.', default='demo')
+parser.add_argument('--directory', '-d', help='Directory to upload, default is current working directory. (' + os.getcwd() + ')', default=os.getcwd())
+args = parser.parse_args()
+
+pw = args.password
+if pw == '-':
+    pw = getpass.getpass()
+
+print('Using server: ' + args.server)
```

Comment:
I'm strongly against using print for any purpose, but especially for an app that can potentially be like a regular unix tool, thus emits nothing upon successful run. I like to use logging.console for this purpose so the output can be suppressed, or redirected to a log file without needing stdout/err redirection of the shell it's running under. But as a first version this is fine, just want to start a discussion on the subject.

Reply:
I went and looked up some info about the logging module and this page contained this information:
https://docs.python.org/3/howto/logging.html
Task you want to perform: Display console output for ordinary usage of a command line script or program
The best tool for the task: print()

For this particular output, I think print() is correct - the user would always want to make sure they didn't accidentally flub the server or repo. But for the other print's in the script, you're correct, and I'll switch those over.